test/CppTests/Tests/BreakpointTests.cs

```diff
@@ -424,6 +424,62 @@ public void ConditionalStringBreakpoints(ITestSettings settings)
             }
         }
 
+        [Theory]
+        [DependsOnTest(nameof(CompileKitchenSinkForBreakpointTests))]
+        [RequiresTestSettings]
+        // lldb-mi does not support -break-watch
+        [UnsupportedDebugger(SupportedDebugger.Lldb, SupportedArchitecture.x64 | SupportedArchitecture.x86)]
+        public void DataBreakpointTest(ITestSettings settings)
+        {
+            this.TestPurpose("Tests that data breakpoints work");
+            this.WriteSettings(settings);
+
+            IDebuggee debuggee = SinkHelper.Open(this, settings.CompilerSettings, DebuggeeMonikers.KitchenSink.Breakpoint);
+
+            using (IDebuggerRunner runner = CreateDebugAdapterRunner(settings))
+            {
+                runner.Launch(settings.DebuggerSettings, debuggee, "-fCalling");
+
+                SourceBreakpoints callingBreakpoints = debuggee.Breakpoints(SinkHelper.Calling, 15);
+                runner.SetBreakpoints(callingBreakpoints);
+
+                runner.Expects.HitBreakpointEvent(SinkHelper.Calling, 15)
+                            .AfterConfigurationDone();
+
+
+                this.Comment("Verify breakpoint condition is met");
+                using (IThreadInspector inspector = runner.GetThreadInspector())
+                {
+                    IFrameInspector mainFrame = inspector.Stack.First();
+                    mainFrame.GetVariable("total");
+
+                    this.Comment("Get DataBreakpointInfo on 'total'");
+                    var response = runner.DataBreakpointInfo("total");
```

Comment:
>response [](http://example.com/codeflow?start=24&length=8)

Verify the `Description` and `AccessTypes` in the response in some way?

Reply:
I guess we should also verify that dataId isn't null or empty.

Reply:
Also maybe verify the DataId is ok (not null)